### PR TITLE
More example descriptions

### PR DESCRIPTION
**Pull request by midanosi**
Had some more Example descriptions in progress!
Hope these aren't too verbose. I'm basically just going through the code to understand how it works, describing that, then in some cases highlighting the key bits from the code that are achieving what the example is doing.

And I'm sometimes pointing out some threejs or threlte functionality that a newbie (like myself) might not have grasped yet, but would otherwise be a blocker to understanding how a particular example works quickly.

Am in progress on fixing/updating the terrain-physics example and adding a description for that too, but will put that in a separate PR

### CODE REVIEW

File: apps/docs-next/src/content/learn/examples/camera/pointer-lock-controls.mdx [hunk 1]
```diff
@@ -6,18 +6,23 @@ showInSidebar: false
...
 <Example path="camera/pointer-lock-controls" />
 
-### Why you might want to use this
+### Explanation
+
+When the scene is clicked, the pointer is locked to the scene, and now pointer movements will control the angle of the camera in the scene.
 
-From the Pointer Lock API MDN doc:
+1. there is no need to click and drag, like with e.g. OrbitControls.
+2. > Pointer lock lets you access mouse events even when the cursor goes past the boundary of the browser or screen (from MDN doc)
```
Comment (grischaerbe) [inline apps/docs-next/src/content/learn/examples/camera/pointer-lock-controls.mdx:23]: Remove "(from MDN doc)"

Comment (grischaerbe) [inline apps/docs-next/src/content/learn/examples/camera/pointer-lock-controls.mdx:23]: Remove "(from MDN doc)"

File: apps/docs-next/src/content/learn/examples/camera/pointer-lock-controls.mdx [hunk 2]
```diff
@@ -6,18 +6,23 @@ showInSidebar: false
...
 
-> Pointer lock lets you access mouse events even when the cursor goes past the boundary of the browser or screen
+To explain the 2nd point, find a threlte scene which uses OrbitControls for it's camera. Now click and drag the cursor left until you hit the edge of your screen.
```
Comment (grischaerbe) [inline apps/docs-next/src/content/learn/examples/camera/pointer-lock-controls.mdx:25]: Capital T for "Threlte"

Comment (grischaerbe) [inline apps/docs-next/src/content/learn/examples/camera/pointer-lock-controls.mdx:25]: Mark "`OrbitControls`" as code with single backticks

File: apps/docs-next/src/content/learn/examples/geometry/terrain.mdx [hunk 3]
```diff
@@ -4,4 +4,61 @@ title: Terrain with 3D noise
 showInSidebar: false
 ---
 
+### Why
+
+The key to creating "random" yet smooth terrain is using "noise".
+
+To create a terrain map, we want to be able to input the x and y coordinate , and return a height value.
+Therefore we want to utilise 2D noise, and this example uses `createNoise2D` from npm package 'simplex-noise'.
```
Comment (grischaerbe) [inline apps/docs-next/src/content/learn/examples/geometry/terrain.mdx:12]: "and this example uses the function `createNoise2D` from the package `simplex-noise`."

File: apps/docs-next/src/content/learn/examples/geometry/terrain.mdx [hunk 4]
```diff
@@ -4,4 +4,61 @@ title: Terrain with 3D noise
...
+```
+for (let i = 0; i < vertices.length; i += 3) {
+		const x = vertices[i]
+    const y = vertices[i + 1]
+    // @ts-ignore
+    vertices[i + 2] = noise(x/4, y/4)
+  }
+```
+
+Aside: Why `i+=3`?
```
Comment (grischaerbe) [inline apps/docs-next/src/content/learn/examples/geometry/terrain.mdx:45]: Cool addition! Make it a `<Tip type="tip">` ;)

File: apps/docs-next/src/content/learn/examples/geometry/terrain.mdx [hunk 5]
```diff
@@ -4,4 +4,61 @@ title: Terrain with 3D noise
...
+
+```
+<T.Mesh
+	{geometry}
+	rotation.x={DEG2RAD * -90}
+>
+...
+```
+
+6. This example intentionally used only part of the noise map (the middle 25%), in order to generate gentler terrain.
```
Comment (grischaerbe) [inline apps/docs-next/src/content/learn/examples/geometry/terrain.mdx:60]: Make this a `<Tip type="tip">` too 👍

File: apps/docs-next/src/content/learn/examples/animation/transitions.mdx [hunk 6]
```diff
@@ -6,8 +6,29 @@ showInSidebar: false
...
 Transition seamlessly between GLTF animations.
 
-## How it works
+<Example path="animation/transitions" />
 
-The important stuff happens in the `transitionTo` function in `Character.svelte`
+## Explanation
 
-<Example path="animation/transitions" />
+GLTF is a comprehensive file format for 3D models, and it supports animations.
```
Comment (grischaerbe) [inline apps/docs-next/src/content/learn/examples/animation/transitions.mdx:13]: I think the correct capitalization is "glTF" although I didn't follow through with this in the rest of the codebase. Still, it's never too late to start!

File: apps/docs-next/src/content/learn/examples/animation/transitions.mdx [hunk 7]
```diff
@@ -6,8 +6,29 @@ showInSidebar: false
...
+In this example, we extract the animations from the gltf file, play them, and crossfade between them.
+
+### What is the code doing?
+
+1. we extract the variables `gltf` and `actions`;
+
+   `const { gltf, actions } = useGltfAnimations()`
+
+2. bind `gltf` to the GLTF component,
```
Comment (grischaerbe) [inline apps/docs-next/src/content/learn/examples/animation/transitions.mdx:22]:
```md
bind `gltf` to the `<GLTF>` component:
```

File: apps/docs-next/src/content/learn/examples/animation/transitions.mdx [hunk 8]
```diff
@@ -6,8 +6,29 @@ showInSidebar: false
...
-The important stuff happens in the `transitionTo` function in `Character.svelte`
+## Explanation
 
-<Example path="animation/transitions" />
+GLTF is a comprehensive file format for 3D models, and it supports animations.
+In this example, we extract the animations from the gltf file, play them, and crossfade between them.
+
+### What is the code doing?
+
+1. we extract the variables `gltf` and `actions`;
```
Comment (grischaerbe) [inline apps/docs-next/src/content/learn/examples/animation/transitions.mdx:18]:
```md
Extract the variables `gltf` and `actions`:
```

File: apps/docs-next/src/content/learn/examples/animation/transitions.mdx [hunk 9]
```diff
@@ -6,8 +6,29 @@ showInSidebar: false
...
+```
+<GLTF
+  bind:gltf={$gltf}
+  url="https://threejs.org/examples/models/gltf/Xbot.glb"
+/>
+```
+
+this causes `actions` to be populated with an array of the available animations in that gltf file
+
+3. (tip) run `console.log(Object.entries($actions))` to see the available action strings, and the shape of the animation object. By doing this, you'll discover that this example is only using 3 of the 7 available animations attached to this gltf file.
```
Comment (grischaerbe) [inline apps/docs-next/src/content/learn/examples/animation/transitions.mdx:33]: Also make this a `<Tip type="tip">` 👍

**Review by grischaerbe — CHANGES_REQUESTED** [review]
Hey @midanosi, thanks so much for your contribution! It really helps understanding what's going on for beginners and advanced people alike.

I have some remarks, nothing special, otherwise LGTM 🚀 ❤️

**Comment by midanosi** [comment]
> Hey @midanosi, thanks so much for your contribution! It really helps understanding what's going on for beginners and advanced people alike.
> 
> I have some remarks, nothing special, otherwise LGTM 🚀 ❤️

Thank you for the compliment and the thorough feedback, have implemented and committed everything you suggested.
Two extra changes I made on top were startcasing 'threlte' to 'Threlte' in a few other places in the repo, and adding one more <Tip> in the terrain example. (using <Tip> is such a great shout)

**Pull request merged**